### PR TITLE
Add serialization logic for the `NDCubeSequence` and `NDCollection` instances.

### PR DESCRIPTION
**Pull request by ViciousEagle03**
Fixes: #755

### CODE REVIEW

**Comment by ViciousEagle03** [comment]
@braingram, @Cadair, could you guys please review it.

**Comment by Cadair** [comment]
pre-commit.ci autofix

File: ndcube/asdf/converters/ndcollection_converter.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    def from_yaml_tree(self, node, tag, ctx):
+        from ndcube.ndcollection import NDCollection
+
+        key_value_pairs = list(zip(node["keys"], node["value"]))
+        aligned_axes = list(node.get("aligned_axes").values())
+        aligned_axes = tuple(tuple(lst) for lst in aligned_axes)
+        ndcollection = NDCollection(key_value_pairs,
+                            meta=node.get("meta"),
+                            aligned_axes = aligned_axes)
```
Comment (Cadair) [inline ndcube/asdf/converters/ndcollection_converter.py:16]:
```suggestion
                                    meta=node.get("meta"),
                                    aligned_axes = aligned_axes)
```

File: ndcube/asdf/converters/ndcollection_converter.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        return ndcollection
+
+
+    def to_yaml_tree(self, ndcollection, tag, ctx):
+
```
Comment (Cadair) [inline ndcube/asdf/converters/ndcollection_converter.py:21]:
```suggestion
```

File: ndcube/asdf/converters/ndcollection_converter.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        key_value_pairs = list(zip(node["keys"], node["value"]))
+        aligned_axes = list(node.get("aligned_axes").values())
+        aligned_axes = tuple(tuple(lst) for lst in aligned_axes)
+        ndcollection = NDCollection(key_value_pairs,
+                            meta=node.get("meta"),
+                            aligned_axes = aligned_axes)
+        return ndcollection
+
+
```
Comment (Cadair) [inline ndcube/asdf/converters/ndcollection_converter.py:19]:
```suggestion
```

File: ndcube/asdf/converters/ndcubesequence_converter.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    def from_yaml_tree(self, node, tag, ctx):
+        from ndcube.ndcube_sequence import NDCubeSequence
+
+        return NDCubeSequence(node["data"],
+                              meta=node.get("meta"),
+                              common_axis=node.get("common_axis"))
+
+
+    def to_yaml_tree(self, ndcseq, tag, ctx):
+
```
Comment (Cadair) [inline ndcube/asdf/converters/ndcubesequence_converter.py:17]:
```suggestion
```

File: ndcube/asdf/converters/ndcubesequence_converter.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    types = ["ndcube.ndcube_sequence.NDCubeSequence"]
+
+    def from_yaml_tree(self, node, tag, ctx):
+        from ndcube.ndcube_sequence import NDCubeSequence
+
+        return NDCubeSequence(node["data"],
+                              meta=node.get("meta"),
+                              common_axis=node.get("common_axis"))
+
+
```
Comment (Cadair) [inline ndcube/asdf/converters/ndcubesequence_converter.py:15]:
```suggestion
```

File: ndcube/asdf/converters/tests/test_ndcollection_converter.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+def test_serialization_cube(create_ndcollection_cube, tmp_path):
+    ndcollection = create_ndcollection_cube
+    file_path = tmp_path / "test.asdf"
+    with asdf.AsdfFile() as af:
+        af["ndcube_gwcs"] = ndcollection
+        af.write_to(file_path)
+
+    with asdf.open(file_path) as af:
+        assert_collections_equal(af["ndcube_gwcs"], ndcollection)
+@pytest.fixture
```
Comment (Cadair) [inline ndcube/asdf/converters/tests/test_ndcollection_converter.py:33]:
```suggestion


@pytest.fixture
```

**Review by Cadair — APPROVED** [review]

**Pull request merged**